### PR TITLE
Simple JSON-RPC API Documentation

### LINKED ISSUE #1023: Document JSON RPC API
Find somewhere to host the JSON RPC API documentation.  Should be tied to solana branch/tag version

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Closes #1023 
If/as our api grows significantly larger or more complex, we may want to write an OpenAPI document, which would open up options to automatically general HTML documentation. But I think that is more work than it is worth right now.

That said, some aspects of github's markdown flavoring are fairly annoying -- especially the non-existent line wrapping in code blocks. So perhaps that is an argument for hosting this somewhere else.

### CODE REVIEW

**Comment by mvines** [comment]
"Documetation"

**Comment by CriesofCarrots** [comment]
🤦‍♀️

File: doc/json-rpc.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+Requests can be sent in batches by sending an array of JSON-RPC request objects as the data for a single POST.
+
+JSON RPC API Reference
+---
+
+### confirmTransaction
+Returns a transaction receipt
+
+##### Parameters:
+* `string` - SIGNATURE of transaction to confirm, as base-58 encoded string
```
Comment (mvines) [inline doc/json-rpc.md]: SIGNATURE/PUBKEY/HASH - there's a little bit of yelling in this doc.  Perhaps we define these words at the top of the doc then use them as proper nouns throughout the rest of the doc:  Signature/PublicKey/Hash

File: doc/json-rpc.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+---
+
+### getBalance
+Returns the balance of the account of provided pubkey
+
+##### Parameters:
+* `string` - PUBKEY of account to query, as base-58 encoded string
+
+##### Results:
+* `integer` - quantity, as a signed 64-bit integer i64
```
Comment (mvines) [inline doc/json-rpc.md]: signed?  The balance can't go negative right?

I'd drop the Rust type reference (`i64`), there's an `u64` in the doc too.   I don't think that's helpful in the JSON context

File: doc/json-rpc.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+{"jsonrpc":"2.0","result":268,"id":1}
+```
+
+---
+
+### sendTransaction
+Creates new transaction
+
+##### Parameters:
+* `array` - array of octets [u8] containing a fully-signed TRANSACTION
```
Comment (mvines) [inline doc/json-rpc.md]: Do we document the actual transaction octet format here or elsewhere?

File: doc/json-rpc.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+Solana nodes accept HTTP requests using the [JSON-RPC 2.0](https://www.jsonrpc.org/specification) specification.
+
+To interact with a Solana node inside a JavaScript application, use the [solana-web3.js](https://github.com/solana-labs/solana-web3.js) library, which gives a convenient interface for the RPC methods.
+
+RPC Endpoint
+---
+
+**Default port:** 8899  
+eg. http://localhost:8899, 192.168.1.88:8899
```
Comment (mvines) [inline doc/json-rpc.md]: I'd add `http://` in front of "192.168.1.88:8899"

File: doc/json-rpc.md (hunk 2 above)
Comment (CriesofCarrots) [reply doc/json-rpc.md]: That's a great question. I don't know if there is a use case for a negative balances being accommodated, but they are definitely stored in the Account struct as i64. https://github.com/solana-labs/solana/blob/0c8eec2563e19db24daad0f579e7e77b0e605f7e/src/bank.rs#L70

Fair point about dropping the Rust types.

File: doc/json-rpc.md (hunk 1 above)
Comment (CriesofCarrots) [reply doc/json-rpc.md]: Like it.

File: doc/json-rpc.md (hunk 4 above)
Comment (CriesofCarrots) [reply doc/json-rpc.md]: ✔️

File: doc/json-rpc.md (hunk 3 above)
Comment (CriesofCarrots) [reply doc/json-rpc.md]: It could arguably go on here. Maybe in the new definitions section.

**Pull request merged**